### ISSUE TITLE
Langchain & Langsmith integration for UpTrain

### ISSUE THREAD

**Issue by sourabhagr** [uptrain-ai/uptrain#388]
The idea is to provide a single-line integration for anyone using Langchain & Langsmith to be able to evaluate their LLM applications using UpTrain

**Comment by sky-2002** [comment]
@sourabhagr I can take this up, does this follow similar pattern as the llamaindex integration [here](https://github.com/uptrain-ai/uptrain/blob/main/uptrain/integrations/llama_index.py) or if you could elaborate a little?

**Comment by ashish-1600** [comment]
yes @sky-2002 . It should follow a similar pattern like how we are doing it in llamaindex integration.

**Comment by rumbleFTW** [comment]
@sky-2002 Did you make any progress with this? Else I can take up this Issue and work on it.

**Comment by sky-2002** [comment]
@rumbleFTW Sure, Please go ahead

**Comment by rachittshah** [comment]
@ashish-1600 the llamaindex file allow you to only use the EvalLlamaIndex class to evaluate using Llamaindex.

The correct way to do this is using the [debughandler](https://docs.llamaindex.ai/en/latest/examples/callbacks/LlamaDebugHandler.html) for llamaindex and [Callback handler](https://python.langchain.com/docs/modules/callbacks/) for Langchain.

@sourabhagr I can pick this up as discussed before on our call!
